src/west/main.py

```diff
@@ -519,6 +520,14 @@ def main(argv=None):
         sys.exit(cce.returncode)
     except CommandError as ce:
         sys.exit(ce.returncode)
+    except BadExternalCommand:
+        log.err('external command', args.command,
+                'was improperly defined and could not be run')
+        if args.verbose:
+            raise
+        else:
+            log.inf(for_stack_trace)
+            sys.exit(1)
```

Comment:
This code give different flows, depending on the `verbose` setting.
I don't think it's common for `-v` to change flow of the program, only to give higher level of verbosity.
Can you explain a bit more, why ?

Reply:
I disagree with the assessment that it changes the flow of the program. I think it really does just give a higher level of verbosity.

Since this is the top level main() function, the only difference in behavior is whether the stack trace is printed or the short error message is. Either way, after the diagnostic message (trace or error), the program exits immediately with a nonzero exit code. So I think this is just verbosity.

It exists elsewhere in the exception handlers in this same try block btw.

Reply:
I think it changes the flow, as we'll do a `sys.exit()` right here, as in principle we can't know what higher level will do with the exception.

But ok to keep as-is.

> It exists elsewhere in the exception handlers in this same try block btw.

Maybe I just didn't notice before.

Reply:
I've rewritten this in a different way to make that more clear using the traceback module.

Reply:
> I think it changes the flow, as we'll do a `sys.exit()` right here, as in principle we can't know what higher level will do with the exception.

In this case we do, because we are main, and the higher layer is the python interpreter itself and its behavior when an exception is not handled is well known.

Even if we weren't the higher layer, it's also worth noting that `sys.exit()` doesn't actually abort the program -- it raises SystemExit. So even in the (unsupported and not stable API) case where someone imported main and ran it, we're just trading that exception for another.

At any rate, I agree that this point is worth making explicit in the code. Thanks again for the review!